kamailio/websockets-role.cfg

```diff
@@ -62,65 +62,83 @@ route[NAT_WEBSOCKETS_CORRECT]
     }
 }
 
+route[WEBSOCKETS_ORIGIN_IP]
+{
+    if ($hdr(X-Forwarded-For) == $null) {
+        xlog("L_INFO", "websocket|log|request from has no X-Forwarded-For, originates from $si\n");
+        $var(orig_ip) = $si;
+    } else if ($si =~ "MY_WEBSOCKET_PROXY_IP_ADDRESS") {
+        xlog("L_INFO", "websocket|log|request X-Forwarded-For $hdr(X-Forwarded-For) from $si\n");
+        $var(orig_ip) = $hdr(X-Forwarded-For);
```

Comment:
whatif `X-Forwarded-For` is sent but `MY_WEBSOCKET_PROXY_IP_ADDRESS` matches ?

Reply:
@lazedo I'm not sure how to change the value at runtime. Can you explain what I'm looking for in the presence-role?

Reply:
Think need to use ACL for this.
This because HA proxy may be also have redurdancy like in Amazon.